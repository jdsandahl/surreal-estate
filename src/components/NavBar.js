import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import "../styles/NavBar.css";
import Logo from "../images/real-estate-logo.png";
import FacebookLogin from "react-facebook-login";
import ErrorBoundary from "./ErrorBoundary";

const NavBar = ({ userId, onLogin, onLogout }) => {
  return (
    <div className="navbar">
      <img className="logo" src={Logo} alt="logo" />
      <h2 className="navbar-title">Surreal Estate</h2>
      <ul className="navbar-links">
        <li className="navbar-links-item">
          <Link to="/" className="navbar-link">
            View Properties
          </Link>
        </li>
        {userId && (
          <li className="navbar-links-item">
            <Link to="saved-properties" className="navbar-link">
              Saved Properties
            </Link>
          </li>
        )}
        <li className="navbar-links-item">
          <Link to="add-property" className="navbar-link">
            Add Property
          </Link>
        </li>

        <li className="navbar-links-item">
          {!userId ? (
            <ErrorBoundary>
              <FacebookLogin
                appId="206769633967359"
                autoLoad={false}
                callback={onLogin}
                cssClass="navbar__login"
                icon="fa-facebook"
              />
            </ErrorBoundary>
          ) : (
            <button className="navbar__login" onClick={onLogout}>
              Sign Out
            </button>
          )}
        </li>
      </ul>
    </div>
  );
};

NavBar.propTypes = {
  userId: PropTypes.string,
  onLogin: PropTypes.func,
  onLogout: PropTypes.func,
};

export default NavBar;
