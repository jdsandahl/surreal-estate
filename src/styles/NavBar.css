.navbar {
    background-color: #ffffff;
    border-bottom: 1px solid #c6c6c6;
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-start;
}

.logo {
    height: 50px;
    padding: 10px;
}

.navbar-links {
    list-style: none;
}

.navbar-links-item {
    display: inline-block;
    margin: none;
    padding: 10px;
    text-decoration: none;
    color: black;
}
