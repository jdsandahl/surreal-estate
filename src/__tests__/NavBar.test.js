import React from "react";
import { render } from "@testing-library/react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import NavBar from "../components/NavBar";

describe("NavBar", () => {
  test('the page renders correctly against snapshot', () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <Router history={history}>
        <NavBar />
      </Router>
    );
    const component = asFragment();

    expect(component).toMatchSnapshot();
  });

  test("renders NavBar component", () => {
    const history = createMemoryHistory();
    const { getByText, getByAltText } = render(
      <Router history={history}>
        <NavBar />
      </Router>
    );

    const logoElement = getByAltText(/logo/i);
    const titleElement = getByText(/surreal estate/i);

    expect(titleElement).toBeInTheDocument();
    expect(logoElement).toBeInTheDocument();
    expect(getByText("View Properties")).toBeInTheDocument();
    expect(getByText("Add Property")).toBeInTheDocument();
  });
});
