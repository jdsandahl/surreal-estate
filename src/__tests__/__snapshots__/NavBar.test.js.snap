// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar the page renders correctly against snapshot 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <img
      alt="logo"
      class="logo"
      src="real-estate-logo.png"
    />
    <h2>
      Surreal Estate
    </h2>
    <ul
      class="navbar-links"
    >
      <li
        class="navbar-links-item"
      >
        <a
          class="navbar-link"
          href="/"
        >
          View Properties
        </a>
      </li>
      <li
        class="navbar-links-item"
      >
        <a
          class="navbar-link"
          href="/add-property"
        >
          Add Property
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
